### PR TITLE
Enable osx_arm64 for ilp64 via conda_build_config.yaml

### PR DESCRIPTION
**Pull request by mkitti**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Revisits #105

### CODE REVIEW

**Comment by mkitti** [comment]
@conda-forge-admin, please rerender

**Comment by mkitti** [comment]
cc: @ngam

**Comment by mkitti** [comment]
huh, this may actually work

**Comment by h-vetinari** [comment]
I don't have anything against this in principle, but with the lib names & infrastructure on osx-arm (which I don't feel I know well enough), I'm going to defer to @isuruf for this one.

**Comment by mkitti** [comment]
@h-vetinari , if you do know the other platforms, could you check that the `OBJCONV=objconv` `make` parameter does not create other issues?

**Comment by isuruf** [comment]
I now remember the last time I tried this. `objconv` doesn't seem to support arm64 binaries. I'm not sure the artifact produced would work.

**Comment by mkitti** [comment]
Upstream Julia seems to use vanilla `objconv`: https://github.com/JuliaLang/julia/blob/master/deps/objconv.mk
https://github.com/JuliaLang/julia/blob/91f068c5c219275f1115056084417057a66240b7/deps/openblas.mk#L37
https://github.com/JuliaPackaging/Yggdrasil/blob/fd160480030edfa8eb7dd21e19db2e7aff323a2f/O/OpenBLAS/common.jl#L68

Perhaps it does work?

**Comment by mkitti** [comment]
> Could you add the following to conda-forge.yml in the root of the repo to see what it produces for osx-arm64?

Done

**Comment by mkitti** [comment]
Does this work?

**Comment by isuruf** [comment]
objconv doesn't seem to support CPU_TYPE_ARM64 if I read the source correctly at https://github.com/staticfloat/objconv/blob/master/src/macho.cpp#L18-L23

**Comment by mkitti** [comment]
That mirror of objconv is quite old. That said I know @staticfloat this figured out for Julia CI, so perhaps he could chime in.

**Comment by mkitti** [comment]
I just checked the latest source code from https://www.agner.org/optimize/ and those lines are exactly the same:
```
// Machine names
SIntTxt MacMachineNames[] = {
   {MAC_CPU_TYPE_I386,      "Intel 32 bit"},
   {MAC_CPU_TYPE_X86_64,    "Intel 64 bit"},
   {MAC_CPU_TYPE_ARM,       "Arm"},
   {MAC_CPU_TYPE_SPARC,     "Sparc"},
   {MAC_CPU_TYPE_POWERPC,   "Power PC 32 bit"},
   {MAC_CPU_TYPE_POWERPC64, "Power PC 64 bit"}
};
```

Perhaps the ARM support is generic?

Julia builds openblas-ilp64 just fine with version 2.49 of objconv:
https://github.com/JuliaPackaging/Yggdrasil/blob/32d14ffa78c8f3deeb729cc1f93e53b3aebf6be3/O/Objconv/build_tarballs.jl#L10-L11

That's the exact same version in conda-forge:
https://github.com/conda-forge/objconv-feedstock/blob/3dab785a41bebd0e91982441eac9c46341045dad/recipe/meta.yaml#L2

The objconv change log is as follows
```
2022-04-28 version 2.53
* add disassembler support for AVX512-FP16, etc.

2020-06-08 version 2.52
* fix  problem in asm syntax output for lea

2018-10-08 version 2.51
* fix bug in movq mm0,rax

2018-08-15 version 2.50
* minor change in omf2cof

2018-01-29 version 2.49
* fixed minor problem with disassembly of DLL
```

**Comment by mkitti** [comment]
Update: I spoke with Elliott Saba and https://github.com/staticfloat/objconv is now the latest version at v2.53. Perhaps the conda-forge version will update soon?

**Comment by mkitti** [comment]
> The update didn't improve the issue with compiling natively on m1 fwiw, not sure if it is related after all..

Did you switch back to the main branch of the julia-feedstock to try to compile with `USE_SYSTEM_BLAS=1` ?
https://github.com/conda-forge/julia-feedstock/blob/7fb588639bb7a857c91e5035423657dd5526c3e6/recipe/build.sh#L45

**Comment by mkitti** [comment]
I'm not sure I understand the Travis CI failure here? It stalled.

**Comment by mkitti** [comment]
@isuruf, Also where are we with this at the moment? I have verified that the `objconv` we are using is the exact same `objconv` that Julia uses to build `openblas`. That build seems to work.

Since Julia is the main dependency that needs this at all, could we proceed and see we can use this package to at least build the core parts of Julia?

**Comment by mkitti** [comment]
> The update didn't improve the issue with compiling natively on m1 fwiw, not sure if it is related after all..

Did you change the settings back to the main branch? Is there anything in the log?

I don't think the update should have. It shouldn't have changed anything. It was just to make absolutely sure that we're already using the same version that upstream Julia is using.

**Comment by mkitti** [comment]
I'm on my Mac Studio at the moment. How do I download this to test it?

**Comment by isuruf** [comment]
We can give this a try.

**Comment by mkitti** [comment]
Great. Thank you @isuruf .

**Comment by mkitti** [comment]
@rgommers @isuruf @ngam To make absolutely sure that the issue is bumping fortran compiler version rather than the other changes here, is there a way to obtain a build from before the merge commit but with the change in fortran?

File: recipe/build.sh [hunk 1]
```diff
@@ -60,6 +60,11 @@ fi
 
 export HOSTCC=$CC_FOR_BUILD
 
+OBJCONV=""
+if [[ "${target_platform}" == "osx-arm64" && "${build_platform}" != "osx-arm64" ]]; then
```
[the remark this answers is not in the document]
Comment (mkitti) [reply recipe/build.sh:64]: This just means that we're cross-compiling, right?

**Comment by rgommers** [comment]
Thanks for following up, and for opening a new PR for testing

> @isuruf could you please double-check on the naming and suffix changes @mkitti made?

The contents of this PR look good to me, and if there was any issue around suffix or similar, we'd see much more breakage I'd think.

**Pull request merged**